engine/enginerpc/engine_impl.cc

```diff
@@ -234,24 +236,226 @@ grpc::Status engine_impl::GetHost(grpc::ServerContext* context [[maybe_unused]],
                                   [[maybe_unused]],
                                   EngineHost* response) {
   std::string err;
-  auto fn = std::packaged_task<int(void)>(
-      [&err, request, host = response]() -> int32_t {
-        std::shared_ptr<com::centreon::engine::host> selectedhost;
-        std::tie(selectedhost, err) = get_host(*request);
-        if (!err.empty()) {
-          return 1;
-        }
 
-        host->set_name(selectedhost->name());
-        host->set_alias(selectedhost->get_alias());
-        host->set_address(selectedhost->get_address());
-        host->set_check_period(selectedhost->check_period());
-        host->set_current_state(
-            static_cast<EngineHost::State>(selectedhost->get_current_state()));
-        host->set_id(selectedhost->host_id());
-        return 0;
-      });
+  auto fn = std::packaged_task<int(void)>([&err, request,
+                                           host = response]() -> int32_t {
+    std::shared_ptr<com::centreon::engine::host> selectedhost;
+    std::tie(selectedhost, err) = get_host(*request);
 
+    if (!err.empty()) {
+      return 1;
+    }
+    // locals
+    std::string cg_str;
+    std::string c_str;
+    std::string p_str;
+    std::string child_str;
+    std::string services_str;
+    std::string notifications_str;
+    std::string groups_name_str;
+    std::string state_history_str;
+    std::string custom_variables_str;
+    std::ostringstream oss;
+    hostgroup* hg{selectedhost->get_parent_groups().front()};
+
+    if (selectedhost->get_contactgroups().empty())
+      cg_str = "\"nullptr\"";
```

Comment:
not needed, with the new option available in ccc, we can force the display of an empty field.

Reply:
done